### ISSUE TITLE
0.4.3: failed to add UI from resource /im/dino/Dino/unified_main_content.ui: .:26:1 Invalid object type 'DinoUiConversationSelector'

### ISSUE THREAD

**Issue by 0-wiz-0** [dino/dino#1585]
I've tried to update the dino package in pkgsrc to version 0.4.3.
I had to apply this patch to get it to build at all:
```
Fix build with vala 0.56
https://github.com/dino/dino/issues/1576

--- plugins/gpgme-vala/vapi/gpgme_public.vapi.orig      2024-05-15 10:07:26.045175155 +0000
+++ plugins/gpgme-vala/vapi/gpgme_public.vapi
@@ -22,9 +22,9 @@ public class Key {
     public string issuer_name;
     public string chain_id;
     public Validity owner_trust;
-    [CCode(array_null_terminated = true)]
+    [CCode (array_length = false, array_null_terminated = true)]
     public SubKey[] subkeys;
-    [CCode(array_null_terminated = true)]
+    [CCode (array_length = false, array_null_terminated = true)]
     public UserID[] uids;
     public KeylistMode keylist_mode;
     // public string fpr; // requires gpgme >= 1.7.0
```
It still shows _lots_ of warnings but compiles. However, it doesn't start with
```
failed to add UI from resource /im/dino/Dino/unified_main_content.ui: .:26:1 Invalid object type 'DinoUiConversationSelector'
```
I have no idea what that wants to tell me. Can you please advise?
I guess it's caused by a missing dependency? Or is the installation incomplete?

**Comment by licaon-kter** [comment]
maybe try to build this branch instead https://github.com/dino/dino/commits/v0.4/ ?

**Comment by 0-wiz-0** [comment]
Thanks for the tip. Using that, I don't need the patch above, but I still get the same error when starting dino.

**Comment by licaon-kter** [comment]
Did you clone fresh or just checkout-ing commits? Maybe try from a clean tree

**Comment by 0-wiz-0** [comment]
Fresh clone direct from github.

**Comment by mar-v-in** [comment]
Do you still have the patch `patch-main_CMakeLists.txt` previously used in pkgsrc active? That was to fix the linking issue causing the error you see.

**Comment by 0-wiz-0** [comment]
Thank you, @mar-v-in , that was it. The patch didn't apply cleanly and linked to a closed ticket (#438) so I thought it had been applied. When I add that again, dino starts up.

**Comment by mar-v-in** [comment]
The patch is merged on master but not in 0.4.x. I will add it to 0.4.4 so it won't be needed after that anymore (same as the other patch you currently have to add on 0.4.3)
